test/functional/editor/K_spec.lua

```diff
@@ -33,6 +34,10 @@ describe('K', function()
     feed('i'..test_file..'<ESC>K')
     retry(nil, nil, function() eq(1, eval('filereadable("'..test_file..'")')) end)
     eq({'fnord'}, eval("readfile('"..test_file.."')"))
+    -- Confirm that Neovim is still in terminal mode after K is pressed (#16692).
+    eq('t', eval('mode()'))
+    feed('<space>')  -- Any key, not just <space>, can be used here to escape.
+    retry(nil, 5000, function() eq('n', eval('mode()')) end)
```

Comment:
I don't think this changes anything, as the `<Space>` is only pressed once.

Reply:
The `eval` is always processed after the `feed`, so this `retry` doesn't do anything.

Reply:
Would the call to `eval` force processing of pending items to complete prior to its own execution?

Reply:
Events are only processed when there are no characters available for input:
<https://github.com/neovim/neovim/blob/70fe3ce004f1c17328257349da5528757fdcd354/src/nvim/state.c#L41-L60>